core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -314,6 +316,16 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
         final ProxiedAddresses proxiedAddresses = determineProxiedAddresses(channel, headers);
         final InetAddress clientAddress = config.clientAddressMapper().apply(proxiedAddresses).getAddress();
 
+        // Handle max connection age for HTTP/1.
+        if (responseEncoder instanceof ServerHttp1ObjectEncoder &&
+            ((ServerHttp1ObjectEncoder) responseEncoder).isClosedConnection()) {
+            final ServiceRequestContext reqCtx =
+                    newEarlyRespondingRequestContext(channel, req, hostname, virtualHost,
+                                                     proxiedAddresses, clientAddress, null);
+            respond(ctx, reqCtx, HttpStatus.INTERNAL_SERVER_ERROR, null, ClosedSessionException.get());
```

Comment:
Isn't it better closing the connection immediately? It could have been nicer if there a standard status code assigned for this case...